### PR TITLE
feat: `I click to perform an advanced lookup on '(.*)' lookup` step

### PR DESCRIPTION
**Pull request by tdashworth**
## Purpose
Added a step to open the advanced lookup dialog for a lookup field. 

## Approach
Uses the driver and CSS selector to find the link to click. 

> Ideally this would be EasyRepro code. 

## TODOs
- [x] Automated test coverage for new code
- [x] Documentation updated (if required)
- [x] Build and tests successful

### CODE REVIEW

File: bindings/src/Capgemini.PowerApps.SpecFlowBindings/Steps/LookupSteps.cs [hunk 1]
```diff
@@ -124,5 +124,11 @@ public static void WhenISelectARelatedLookupInTheForm(string lookupName)
             Driver.WaitUntilAvailable(By.XPath(AppElements.Xpath[AppReference.Entity.LookupFieldExistingValue].Replace("[NAME]", lookupName))).Click();
             Driver.WaitForTransaction();
         }
+
```
Comment (osagiestar) [inline bindings/src/Capgemini.PowerApps.SpecFlowBindings/Steps/LookupSteps.cs:127]: /// <summary>
        /// Open the record by clicking on the Advanced lookup link.
        /// </summary>
        /// <param name="lookupField">The name of the lookup.</param>

Comment (osagiestar) [reply bindings/src/Capgemini.PowerApps.SpecFlowBindings/Steps/LookupSteps.cs:127]: Just a little thought: we could add comment to keep consistency with the rest of the steps and for clarity?

**Comment by tdashworth** [comment]
**Manual validation requested**
Please perform any checks required on https://sfb-pr-8092.crm11.dynamics.com/ then approve https://dev.azure.com/capgeminiuk/GitHub%20Support/_build/results?buildId=8092.

File: bindings/src/Capgemini.PowerApps.SpecFlowBindings/Steps/LookupSteps.cs (hunk 1 above)
Comment (tdashworth) [reply bindings/src/Capgemini.PowerApps.SpecFlowBindings/Steps/LookupSteps.cs:127]: Ah yes!

**Review by osagiestar — APPROVED** [review]
Nice job buddy

**Comment by tdashworth** [comment]
**Manual validation requested**
Please perform any checks required on https://sfb-pr-8093.crm11.dynamics.com/ then approve https://dev.azure.com/capgeminiuk/GitHub%20Support/_build/results?buildId=8093.

**Comment by ewingjm** [comment]
@tdashworth what was the need for the departure from the existing binding that doesn't require the lookup logical name? I don't think it should be possible for two lookup results dialogs (where the `Advanced Lookup` button is) to be on screen at the same time.

**Comment by tdashworth** [comment]
> @tdashworth what was the need for the departure from the existing binding that doesn't require the lookup logical name? I don't think it should be possible for two lookup results dialogs (where the `Advanced Lookup` button is) to be on screen at the same time.

Which is the existing binding? For context, we wrote this new binding because there was a test for it that was missing a binding. The reason for the lookup name was only to assist with the `By.CssSelector()`.

**Comment by ewingjm** [comment]
> > @tdashworth what was the need for the departure from the existing binding that doesn't require the lookup logical name? I don't think it should be possible for two lookup results dialogs (where the `Advanced Lookup` button is) to be on screen at the same time.
> 
> Which is the existing binding? For context, we wrote this new binding because there was a test for it that was missing a binding. The reason for the lookup name was only to assist with the `By.CssSelector()`.

Ah, got it. Just got confused seeing that there was already a step so assumed something had been implemented. Although I did think that I made all the EasyRepro `Lookup` methods compatible with the Advanced Lookups dialog.

**Review by ewingjm — APPROVED** [review]

**Pull request merged**